File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/postal/SavePostalIrregularCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.postal;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.FrameworkCapsuleDto;
import net.seijishikin.jp.normalize.manage.kanrensha.entity.AddressPostalIrregularEntity;

/**
 * 郵便番号不規則更新内容Dto
 */
public class SavePostalIrregularCapsuleDto extends FrameworkCapsuleDto implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 郵便番号不規則Entity */
    private AddressPostalIrregularEntity addressPostalIrregularEntity;

    /**
     * 郵便番号不規則Entityを取得する
     *
     * @return 郵便番号不規則Entity
     */
    public AddressPostalIrregularEntity getAddressPostalIrregularEntity() {
        return addressPostalIrregularEntity;
    }

    /**
     * 郵便番号不規則Entityを設定する
     *
     * @param addressPostalIrregularEntity 郵便番号不規則Entity
     */
    public void setAddressPostalIrregularEntity(final AddressPostalIrregularEntity addressPostalIrregularEntity) {
        this.addressPostalIrregularEntity = addressPostalIrregularEntity;
    }

}

File: app/front/src/components/main/dto/address_postal/savePostalIrregularCapsuleDto.ts
import { FrameworkCapsuleDto, type FrameworkCapsuleDtoInterface } from "seijishikin-jp-normalize_common-tool";
import { AddressPostalIrregularEntity, type AddressPostalIrregularEntityInterface } from "../../entity/addressPostalIrregularEntity";

interface SavePostalIrregularCapsuleDtoInterface extends FrameworkCapsuleDtoInterface {
    /** 郵便番号不規則Entity */
    addressPostalIrregularEntity: AddressPostalIrregularEntityInterface;
}


class SavePostalIrregularCapsuleDto extends FrameworkCapsuleDto implements SavePostalIrregularCapsuleDtoInterface {

    /** 郵便番号不規則Entity */
    addressPostalIrregularEntity: AddressPostalIrregularEntityInterface;

    constructor() {
        super();
        this.addressPostalIrregularEntity = new AddressPostalIrregularEntity();
    }
}

export { type SavePostalIrregularCapsuleDtoInterface, SavePostalIrregularCapsuleDto }
